mk/mockery.mk

```diff
@@ -0,0 +1,12 @@
+MOCKERY_VERSION := 2.36.1
+
+export GO_OUTPUT
+
+.PHONY: install-mockery
+install-mockery: ## Install mockery locally on your GO_OUTPUT (./release) directory
+	mkdir -p $(GO_OUTPUT) && \
+	curl -sSfL https://github.com/vektra/mockery/releases/download/v$(MOCKERY_VERSION)/mockery_$(MOCKERY_VERSION)_$(shell uname -s)_$(shell uname -m).tar.gz | tar -xz -C $(GO_OUTPUT) mockery
+
+.PHONY: mock
+mock: ## Regenerate mocks
```

Comment:
if you drop the .PHONY, and name the make target the name of the resulting binary:
```
$(GO_OUTPUT)/mockery: 
	mkdir -p $(GO_OUTPUT) && \
	curl -sSfL https://github.com/vektra/mockery/releases/download/v$(MOCKERY_VERSION)/mockery_$(MOCKERY_VERSION)_$(shell uname -s)_$(shell uname -m).tar.gz | tar -xz -C $(GO_OUTPUT) mockery
```

Then you can set it to be a dependency on 'mock':
```
.PHONY: mock
mock:  $(GO_OUTPUT)/mockery
```

now when you just run 'make mock', it will install it if needed (and skip the install step if not).

Reply:
oooh thanks, this is great! will update